cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -57,6 +64,12 @@ protected void configure() {
         expose(LineageWriter.class);
         bind(DatasetFramework.class).to(LineageWriterDatasetFramework.class);
         expose(DatasetFramework.class);
+        if (kafkaRequired) {
```

Comment:
why is this in the datasets module? The publishing is a function of the metadata service, not the data set framework.